lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -113,11 +111,16 @@ def #{configuration[:column]}=(position)
               attr_accessible :#{configuration[:column]}
             end
 
+            before_validation :check_top_position
+            
             before_destroy :reload_position
             after_destroy :decrement_positions_on_lower_items
+            
             before_update :check_scope
             after_update :update_positions
-            before_validation :check_top_position
+
+            after_save '@scope_changed = false'
```

Comment:
Should we now remove this line? I agree with your approach of setting `@scope_changed` to `false` straight after adding the item to the list.